{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "007e3d60-363a-42d2-aa02-48169e878005",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── \u001b[1mAttaching packages\u001b[22m ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.6     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.8\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.2.0     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.2     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(tidyverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c2ca8827-bb75-49e9-bc93-31427982b72b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: S4Vectors\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    first, rename\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:tidyr’:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    collapse, desc, slice\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    reduce\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘matrixStats’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(\"DESeq2\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47ee01ec-c762-482d-9ee5-0ede431d50bc",
   "metadata": {},
   "source": [
    "Select ND and NFKB1 samples to compare"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c0c91b7e-1b28-4a76-a3ee-4673c2f4294a",
   "metadata": {},
   "outputs": [],
   "source": [
    "count_tbl <- read.csv(\n",
    "    \"merge_peaks.raw_count.txt\",\n",
    "    sep=\"\\t\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "eb649777-89d7-4cff-91f9-ca45f2d31133",
   "metadata": {},
   "outputs": [],
   "source": [
    "ND.samples <- colnames(count_tbl)[str_detect(colnames(count_tbl), \"^ATAC.*ND\")]\n",
    "NFKB1.samples <- colnames(count_tbl)[str_detect(colnames(count_tbl), \"^ATAC.*NFKB1\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "b26a653e-6f80-486a-9d75-40476f12e63b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 24</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Count_ATAC188_CB_FR062.1_ND_sB</th><th scope=col>Count_ATAC158_CB_FR052.1_ND_sB</th><th scope=col>Count_ATAC137_CB_FR045.1_ND_sB</th><th scope=col>Count_ATAC149_CB_FR049.1_ND_sB</th><th scope=col>Count_ATAC176_CB_FR058.1_ND_sB</th><th scope=col>Count_ATAC245_CB_FR081.1_ND_sB</th><th scope=col>Count_ATAC152_CB_FR050.1_ND_sB</th><th scope=col>Count_ATAC239_CB_FR079.1_ND_sB</th><th scope=col>Count_ATAC155_CB_FR051.1_ND_sB</th><th scope=col>Count_ATAC146_CB_FR048.1_ND_sB</th><th scope=col>⋯</th><th scope=col>Count_ATAC123_CB_FR040.1_ND_sB</th><th scope=col>Count_ATAC191_CB_FR063.1_ND_sB</th><th scope=col>Count_ATAC185_CB_FR061.1_ND_sB</th><th scope=col>Count_ATAC224_CB_FR075.1_ND_sB</th><th scope=col>Count_ATAC111_CB_FR036.1_NFKB1_sB</th><th scope=col>Count_ATAC295_BG_FR100.1_NFKB1_sB</th><th scope=col>Count_ATAC301_BG_FR102.1_NFKB1_sB</th><th scope=col>Count_ATAC074_CB_FR023.1_NFKB1_sB</th><th scope=col>Count_ATAC304_BG_FR103.1_NFKB1_sB</th><th scope=col>Count_ATAC298_BG_FR101.1_NFKB1_sB</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>chr1:10043-10267</th><td>15</td><td> 7</td><td> 9</td><td>10</td><td> 8</td><td>13</td><td>14</td><td>12</td><td>13</td><td> 8</td><td>⋯</td><td>14</td><td>15</td><td>11</td><td>13</td><td> 9</td><td> 6</td><td> 8</td><td>11</td><td> 8</td><td>12</td></tr>\n",
       "\t<tr><th scope=row>chr1:180704-180974</th><td>17</td><td> 7</td><td>11</td><td> 9</td><td> 5</td><td>16</td><td>12</td><td>20</td><td>12</td><td> 9</td><td>⋯</td><td>12</td><td>11</td><td>14</td><td> 8</td><td>12</td><td> 9</td><td> 9</td><td>13</td><td> 9</td><td>13</td></tr>\n",
       "\t<tr><th scope=row>chr1:181330-181577</th><td>18</td><td>14</td><td> 9</td><td> 7</td><td>13</td><td>17</td><td> 8</td><td>16</td><td>11</td><td>16</td><td>⋯</td><td> 8</td><td>15</td><td>12</td><td>14</td><td>10</td><td> 3</td><td> 0</td><td> 4</td><td> 3</td><td> 2</td></tr>\n",
       "\t<tr><th scope=row>chr1:629820-630076</th><td>51</td><td>46</td><td>55</td><td>54</td><td>53</td><td>37</td><td>51</td><td>56</td><td>53</td><td>54</td><td>⋯</td><td>50</td><td>56</td><td>51</td><td>55</td><td>53</td><td>55</td><td>59</td><td>51</td><td>62</td><td>62</td></tr>\n",
       "\t<tr><th scope=row>chr1:630217-630417</th><td> 1</td><td> 0</td><td> 7</td><td> 1</td><td> 2</td><td> 1</td><td> 0</td><td> 3</td><td> 3</td><td>72</td><td>⋯</td><td> 2</td><td> 1</td><td> 2</td><td> 2</td><td> 0</td><td> 2</td><td> 3</td><td> 4</td><td> 6</td><td> 3</td></tr>\n",
       "\t<tr><th scope=row>chr1:630514-630684</th><td> 0</td><td> 2</td><td>51</td><td> 0</td><td> 3</td><td> 0</td><td> 4</td><td> 5</td><td> 3</td><td> 1</td><td>⋯</td><td> 0</td><td> 1</td><td> 1</td><td> 1</td><td> 1</td><td> 0</td><td> 1</td><td> 1</td><td> 0</td><td> 3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 24\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & Count\\_ATAC188\\_CB\\_FR062.1\\_ND\\_sB & Count\\_ATAC158\\_CB\\_FR052.1\\_ND\\_sB & Count\\_ATAC137\\_CB\\_FR045.1\\_ND\\_sB & Count\\_ATAC149\\_CB\\_FR049.1\\_ND\\_sB & Count\\_ATAC176\\_CB\\_FR058.1\\_ND\\_sB & Count\\_ATAC245\\_CB\\_FR081.1\\_ND\\_sB & Count\\_ATAC152\\_CB\\_FR050.1\\_ND\\_sB & Count\\_ATAC239\\_CB\\_FR079.1\\_ND\\_sB & Count\\_ATAC155\\_CB\\_FR051.1\\_ND\\_sB & Count\\_ATAC146\\_CB\\_FR048.1\\_ND\\_sB & ⋯ & Count\\_ATAC123\\_CB\\_FR040.1\\_ND\\_sB & Count\\_ATAC191\\_CB\\_FR063.1\\_ND\\_sB & Count\\_ATAC185\\_CB\\_FR061.1\\_ND\\_sB & Count\\_ATAC224\\_CB\\_FR075.1\\_ND\\_sB & Count\\_ATAC111\\_CB\\_FR036.1\\_NFKB1\\_sB & Count\\_ATAC295\\_BG\\_FR100.1\\_NFKB1\\_sB & Count\\_ATAC301\\_BG\\_FR102.1\\_NFKB1\\_sB & Count\\_ATAC074\\_CB\\_FR023.1\\_NFKB1\\_sB & Count\\_ATAC304\\_BG\\_FR103.1\\_NFKB1\\_sB & Count\\_ATAC298\\_BG\\_FR101.1\\_NFKB1\\_sB\\\\\n",
       "  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\tchr1:10043-10267 & 15 &  7 &  9 & 10 &  8 & 13 & 14 & 12 & 13 &  8 & ⋯ & 14 & 15 & 11 & 13 &  9 &  6 &  8 & 11 &  8 & 12\\\\\n",
       "\tchr1:180704-180974 & 17 &  7 & 11 &  9 &  5 & 16 & 12 & 20 & 12 &  9 & ⋯ & 12 & 11 & 14 &  8 & 12 &  9 &  9 & 13 &  9 & 13\\\\\n",
       "\tchr1:181330-181577 & 18 & 14 &  9 &  7 & 13 & 17 &  8 & 16 & 11 & 16 & ⋯ &  8 & 15 & 12 & 14 & 10 &  3 &  0 &  4 &  3 &  2\\\\\n",
       "\tchr1:629820-630076 & 51 & 46 & 55 & 54 & 53 & 37 & 51 & 56 & 53 & 54 & ⋯ & 50 & 56 & 51 & 55 & 53 & 55 & 59 & 51 & 62 & 62\\\\\n",
       "\tchr1:630217-630417 &  1 &  0 &  7 &  1 &  2 &  1 &  0 &  3 &  3 & 72 & ⋯ &  2 &  1 &  2 &  2 &  0 &  2 &  3 &  4 &  6 &  3\\\\\n",
       "\tchr1:630514-630684 &  0 &  2 & 51 &  0 &  3 &  0 &  4 &  5 &  3 &  1 & ⋯ &  0 &  1 &  1 &  1 &  1 &  0 &  1 &  1 &  0 &  3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 24\n",
       "\n",
       "| <!--/--> | Count_ATAC188_CB_FR062.1_ND_sB &lt;int&gt; | Count_ATAC158_CB_FR052.1_ND_sB &lt;int&gt; | Count_ATAC137_CB_FR045.1_ND_sB &lt;int&gt; | Count_ATAC149_CB_FR049.1_ND_sB &lt;int&gt; | Count_ATAC176_CB_FR058.1_ND_sB &lt;int&gt; | Count_ATAC245_CB_FR081.1_ND_sB &lt;int&gt; | Count_ATAC152_CB_FR050.1_ND_sB &lt;int&gt; | Count_ATAC239_CB_FR079.1_ND_sB &lt;int&gt; | Count_ATAC155_CB_FR051.1_ND_sB &lt;int&gt; | Count_ATAC146_CB_FR048.1_ND_sB &lt;int&gt; | ⋯ ⋯ | Count_ATAC123_CB_FR040.1_ND_sB &lt;int&gt; | Count_ATAC191_CB_FR063.1_ND_sB &lt;int&gt; | Count_ATAC185_CB_FR061.1_ND_sB &lt;int&gt; | Count_ATAC224_CB_FR075.1_ND_sB &lt;int&gt; | Count_ATAC111_CB_FR036.1_NFKB1_sB &lt;int&gt; | Count_ATAC295_BG_FR100.1_NFKB1_sB &lt;int&gt; | Count_ATAC301_BG_FR102.1_NFKB1_sB &lt;int&gt; | Count_ATAC074_CB_FR023.1_NFKB1_sB &lt;int&gt; | Count_ATAC304_BG_FR103.1_NFKB1_sB &lt;int&gt; | Count_ATAC298_BG_FR101.1_NFKB1_sB &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1:10043-10267 | 15 |  7 |  9 | 10 |  8 | 13 | 14 | 12 | 13 |  8 | ⋯ | 14 | 15 | 11 | 13 |  9 |  6 |  8 | 11 |  8 | 12 |\n",
       "| chr1:180704-180974 | 17 |  7 | 11 |  9 |  5 | 16 | 12 | 20 | 12 |  9 | ⋯ | 12 | 11 | 14 |  8 | 12 |  9 |  9 | 13 |  9 | 13 |\n",
       "| chr1:181330-181577 | 18 | 14 |  9 |  7 | 13 | 17 |  8 | 16 | 11 | 16 | ⋯ |  8 | 15 | 12 | 14 | 10 |  3 |  0 |  4 |  3 |  2 |\n",
       "| chr1:629820-630076 | 51 | 46 | 55 | 54 | 53 | 37 | 51 | 56 | 53 | 54 | ⋯ | 50 | 56 | 51 | 55 | 53 | 55 | 59 | 51 | 62 | 62 |\n",
       "| chr1:630217-630417 |  1 |  0 |  7 |  1 |  2 |  1 |  0 |  3 |  3 | 72 | ⋯ |  2 |  1 |  2 |  2 |  0 |  2 |  3 |  4 |  6 |  3 |\n",
       "| chr1:630514-630684 |  0 |  2 | 51 |  0 |  3 |  0 |  4 |  5 |  3 |  1 | ⋯ |  0 |  1 |  1 |  1 |  1 |  0 |  1 |  1 |  0 |  3 |\n",
       "\n"
      ],
      "text/plain": [
       "                   Count_ATAC188_CB_FR062.1_ND_sB\n",
       "chr1:10043-10267   15                            \n",
       "chr1:180704-180974 17                            \n",
       "chr1:181330-181577 18                            \n",
       "chr1:629820-630076 51                            \n",
       "chr1:630217-630417  1                            \n",
       "chr1:630514-630684  0                            \n",
       "                   Count_ATAC158_CB_FR052.1_ND_sB\n",
       "chr1:10043-10267    7                            \n",
       "chr1:180704-180974  7                            \n",
       "chr1:181330-181577 14                            \n",
       "chr1:629820-630076 46                            \n",
       "chr1:630217-630417  0                            \n",
       "chr1:630514-630684  2                            \n",
       "                   Count_ATAC137_CB_FR045.1_ND_sB\n",
       "chr1:10043-10267    9                            \n",
       "chr1:180704-180974 11                            \n",
       "chr1:181330-181577  9                            \n",
       "chr1:629820-630076 55                            \n",
       "chr1:630217-630417  7                            \n",
       "chr1:630514-630684 51                            \n",
       "                   Count_ATAC149_CB_FR049.1_ND_sB\n",
       "chr1:10043-10267   10                            \n",
       "chr1:180704-180974  9                            \n",
       "chr1:181330-181577  7                            \n",
       "chr1:629820-630076 54                            \n",
       "chr1:630217-630417  1                            \n",
       "chr1:630514-630684  0                            \n",
       "                   Count_ATAC176_CB_FR058.1_ND_sB\n",
       "chr1:10043-10267    8                            \n",
       "chr1:180704-180974  5                            \n",
       "chr1:181330-181577 13                            \n",
       "chr1:629820-630076 53                            \n",
       "chr1:630217-630417  2                            \n",
       "chr1:630514-630684  3                            \n",
       "                   Count_ATAC245_CB_FR081.1_ND_sB\n",
       "chr1:10043-10267   13                            \n",
       "chr1:180704-180974 16                            \n",
       "chr1:181330-181577 17                            \n",
       "chr1:629820-630076 37                            \n",
       "chr1:630217-630417  1                            \n",
       "chr1:630514-630684  0                            \n",
       "                   Count_ATAC152_CB_FR050.1_ND_sB\n",
       "chr1:10043-10267   14                            \n",
       "chr1:180704-180974 12                            \n",
       "chr1:181330-181577  8                            \n",
       "chr1:629820-630076 51                            \n",
       "chr1:630217-630417  0                            \n",
       "chr1:630514-630684  4                            \n",
       "                   Count_ATAC239_CB_FR079.1_ND_sB\n",
       "chr1:10043-10267   12                            \n",
       "chr1:180704-180974 20                            \n",
       "chr1:181330-181577 16                            \n",
       "chr1:629820-630076 56                            \n",
       "chr1:630217-630417  3                            \n",
       "chr1:630514-630684  5                            \n",
       "                   Count_ATAC155_CB_FR051.1_ND_sB\n",
       "chr1:10043-10267   13                            \n",
       "chr1:180704-180974 12                            \n",
       "chr1:181330-181577 11                            \n",
       "chr1:629820-630076 53                            \n",
       "chr1:630217-630417  3                            \n",
       "chr1:630514-630684  3                            \n",
       "                   Count_ATAC146_CB_FR048.1_ND_sB ⋯\n",
       "chr1:10043-10267    8                             ⋯\n",
       "chr1:180704-180974  9                             ⋯\n",
       "chr1:181330-181577 16                             ⋯\n",
       "chr1:629820-630076 54                             ⋯\n",
       "chr1:630217-630417 72                             ⋯\n",
       "chr1:630514-630684  1                             ⋯\n",
       "                   Count_ATAC123_CB_FR040.1_ND_sB\n",
       "chr1:10043-10267   14                            \n",
       "chr1:180704-180974 12                            \n",
       "chr1:181330-181577  8                            \n",
       "chr1:629820-630076 50                            \n",
       "chr1:630217-630417  2                            \n",
       "chr1:630514-630684  0                            \n",
       "                   Count_ATAC191_CB_FR063.1_ND_sB\n",
       "chr1:10043-10267   15                            \n",
       "chr1:180704-180974 11                            \n",
       "chr1:181330-181577 15                            \n",
       "chr1:629820-630076 56                            \n",
       "chr1:630217-630417  1                            \n",
       "chr1:630514-630684  1                            \n",
       "                   Count_ATAC185_CB_FR061.1_ND_sB\n",
       "chr1:10043-10267   11                            \n",
       "chr1:180704-180974 14                            \n",
       "chr1:181330-181577 12                            \n",
       "chr1:629820-630076 51                            \n",
       "chr1:630217-630417  2                            \n",
       "chr1:630514-630684  1                            \n",
       "                   Count_ATAC224_CB_FR075.1_ND_sB\n",
       "chr1:10043-10267   13                            \n",
       "chr1:180704-180974  8                            \n",
       "chr1:181330-181577 14                            \n",
       "chr1:629820-630076 55                            \n",
       "chr1:630217-630417  2                            \n",
       "chr1:630514-630684  1                            \n",
       "                   Count_ATAC111_CB_FR036.1_NFKB1_sB\n",
       "chr1:10043-10267    9                               \n",
       "chr1:180704-180974 12                               \n",
       "chr1:181330-181577 10                               \n",
       "chr1:629820-630076 53                               \n",
       "chr1:630217-630417  0                               \n",
       "chr1:630514-630684  1                               \n",
       "                   Count_ATAC295_BG_FR100.1_NFKB1_sB\n",
       "chr1:10043-10267    6                               \n",
       "chr1:180704-180974  9                               \n",
       "chr1:181330-181577  3                               \n",
       "chr1:629820-630076 55                               \n",
       "chr1:630217-630417  2                               \n",
       "chr1:630514-630684  0                               \n",
       "                   Count_ATAC301_BG_FR102.1_NFKB1_sB\n",
       "chr1:10043-10267    8                               \n",
       "chr1:180704-180974  9                               \n",
       "chr1:181330-181577  0                               \n",
       "chr1:629820-630076 59                               \n",
       "chr1:630217-630417  3                               \n",
       "chr1:630514-630684  1                               \n",
       "                   Count_ATAC074_CB_FR023.1_NFKB1_sB\n",
       "chr1:10043-10267   11                               \n",
       "chr1:180704-180974 13                               \n",
       "chr1:181330-181577  4                               \n",
       "chr1:629820-630076 51                               \n",
       "chr1:630217-630417  4                               \n",
       "chr1:630514-630684  1                               \n",
       "                   Count_ATAC304_BG_FR103.1_NFKB1_sB\n",
       "chr1:10043-10267    8                               \n",
       "chr1:180704-180974  9                               \n",
       "chr1:181330-181577  3                               \n",
       "chr1:629820-630076 62                               \n",
       "chr1:630217-630417  6                               \n",
       "chr1:630514-630684  0                               \n",
       "                   Count_ATAC298_BG_FR101.1_NFKB1_sB\n",
       "chr1:10043-10267   12                               \n",
       "chr1:180704-180974 13                               \n",
       "chr1:181330-181577  2                               \n",
       "chr1:629820-630076 62                               \n",
       "chr1:630217-630417  3                               \n",
       "chr1:630514-630684  3                               "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sample_count_tbl <- count_tbl[ \n",
    "    apply(count_tbl[, append(ND.samples, NFKB1.samples)], 1, function(x) any(as.logical(x))), \n",
    "    append(c(\"Chromosome\", \"Start\", \"End\"), paste0(\"Count_\", append(ND.samples, NFKB1.samples)))\n",
    "]\n",
    "peak_coor_as_name <- paste0(sample_count_tbl$Chromosome, \":\", sample_count_tbl$Start, \"-\", sample_count_tbl$End)\n",
    "sample_count_tbl <- sample_count_tbl[, 4:dim(sample_count_tbl)[2]]\n",
    "rownames(sample_count_tbl) <- peak_coor_as_name\n",
    "head(sample_count_tbl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f43623bb-68b3-4e44-b088-3f1cf4df962e",
   "metadata": {},
   "outputs": [],
   "source": [
    "coldata<-as.data.frame(append(\n",
    "    rep(c(\"ND\"), length(ND.samples)),\n",
    "    rep(c(\"NFKB1\"), length(NFKB1.samples))\n",
    "))\n",
    "rownames(coldata) <- colnames(sample_count_tbl) \n",
    "colnames(coldata) <- \"condition\"\n",
    "coldata$condition <- factor(coldata$condition)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "94f915e5-803c-475d-9669-4688475b25de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 24 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>condition</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Count_ATAC188_CB_FR062.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC158_CB_FR052.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC137_CB_FR045.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC149_CB_FR049.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC176_CB_FR058.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC245_CB_FR081.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC152_CB_FR050.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC239_CB_FR079.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC155_CB_FR051.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC146_CB_FR048.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC206_CB_FR069.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC143_CB_FR047.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC140_CB_FR046.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC194_CB_FR064.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC123_CB_FR040.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC191_CB_FR063.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC185_CB_FR061.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC224_CB_FR075.1_ND_sB</th><td>ND   </td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC111_CB_FR036.1_NFKB1_sB</th><td>NFKB1</td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC295_BG_FR100.1_NFKB1_sB</th><td>NFKB1</td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC301_BG_FR102.1_NFKB1_sB</th><td>NFKB1</td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC074_CB_FR023.1_NFKB1_sB</th><td>NFKB1</td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC304_BG_FR103.1_NFKB1_sB</th><td>NFKB1</td></tr>\n",
       "\t<tr><th scope=row>Count_ATAC298_BG_FR101.1_NFKB1_sB</th><td>NFKB1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 24 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & condition\\\\\n",
       "  & <fct>\\\\\n",
       "\\hline\n",
       "\tCount\\_ATAC188\\_CB\\_FR062.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC158\\_CB\\_FR052.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC137\\_CB\\_FR045.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC149\\_CB\\_FR049.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC176\\_CB\\_FR058.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC245\\_CB\\_FR081.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC152\\_CB\\_FR050.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC239\\_CB\\_FR079.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC155\\_CB\\_FR051.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC146\\_CB\\_FR048.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC206\\_CB\\_FR069.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC143\\_CB\\_FR047.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC140\\_CB\\_FR046.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC194\\_CB\\_FR064.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC123\\_CB\\_FR040.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC191\\_CB\\_FR063.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC185\\_CB\\_FR061.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC224\\_CB\\_FR075.1\\_ND\\_sB & ND   \\\\\n",
       "\tCount\\_ATAC111\\_CB\\_FR036.1\\_NFKB1\\_sB & NFKB1\\\\\n",
       "\tCount\\_ATAC295\\_BG\\_FR100.1\\_NFKB1\\_sB & NFKB1\\\\\n",
       "\tCount\\_ATAC301\\_BG\\_FR102.1\\_NFKB1\\_sB & NFKB1\\\\\n",
       "\tCount\\_ATAC074\\_CB\\_FR023.1\\_NFKB1\\_sB & NFKB1\\\\\n",
       "\tCount\\_ATAC304\\_BG\\_FR103.1\\_NFKB1\\_sB & NFKB1\\\\\n",
       "\tCount\\_ATAC298\\_BG\\_FR101.1\\_NFKB1\\_sB & NFKB1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 24 × 1\n",
       "\n",
       "| <!--/--> | condition &lt;fct&gt; |\n",
       "|---|---|\n",
       "| Count_ATAC188_CB_FR062.1_ND_sB | ND    |\n",
       "| Count_ATAC158_CB_FR052.1_ND_sB | ND    |\n",
       "| Count_ATAC137_CB_FR045.1_ND_sB | ND    |\n",
       "| Count_ATAC149_CB_FR049.1_ND_sB | ND    |\n",
       "| Count_ATAC176_CB_FR058.1_ND_sB | ND    |\n",
       "| Count_ATAC245_CB_FR081.1_ND_sB | ND    |\n",
       "| Count_ATAC152_CB_FR050.1_ND_sB | ND    |\n",
       "| Count_ATAC239_CB_FR079.1_ND_sB | ND    |\n",
       "| Count_ATAC155_CB_FR051.1_ND_sB | ND    |\n",
       "| Count_ATAC146_CB_FR048.1_ND_sB | ND    |\n",
       "| Count_ATAC206_CB_FR069.1_ND_sB | ND    |\n",
       "| Count_ATAC143_CB_FR047.1_ND_sB | ND    |\n",
       "| Count_ATAC140_CB_FR046.1_ND_sB | ND    |\n",
       "| Count_ATAC194_CB_FR064.1_ND_sB | ND    |\n",
       "| Count_ATAC123_CB_FR040.1_ND_sB | ND    |\n",
       "| Count_ATAC191_CB_FR063.1_ND_sB | ND    |\n",
       "| Count_ATAC185_CB_FR061.1_ND_sB | ND    |\n",
       "| Count_ATAC224_CB_FR075.1_ND_sB | ND    |\n",
       "| Count_ATAC111_CB_FR036.1_NFKB1_sB | NFKB1 |\n",
       "| Count_ATAC295_BG_FR100.1_NFKB1_sB | NFKB1 |\n",
       "| Count_ATAC301_BG_FR102.1_NFKB1_sB | NFKB1 |\n",
       "| Count_ATAC074_CB_FR023.1_NFKB1_sB | NFKB1 |\n",
       "| Count_ATAC304_BG_FR103.1_NFKB1_sB | NFKB1 |\n",
       "| Count_ATAC298_BG_FR101.1_NFKB1_sB | NFKB1 |\n",
       "\n"
      ],
      "text/plain": [
       "                                  condition\n",
       "Count_ATAC188_CB_FR062.1_ND_sB    ND       \n",
       "Count_ATAC158_CB_FR052.1_ND_sB    ND       \n",
       "Count_ATAC137_CB_FR045.1_ND_sB    ND       \n",
       "Count_ATAC149_CB_FR049.1_ND_sB    ND       \n",
       "Count_ATAC176_CB_FR058.1_ND_sB    ND       \n",
       "Count_ATAC245_CB_FR081.1_ND_sB    ND       \n",
       "Count_ATAC152_CB_FR050.1_ND_sB    ND       \n",
       "Count_ATAC239_CB_FR079.1_ND_sB    ND       \n",
       "Count_ATAC155_CB_FR051.1_ND_sB    ND       \n",
       "Count_ATAC146_CB_FR048.1_ND_sB    ND       \n",
       "Count_ATAC206_CB_FR069.1_ND_sB    ND       \n",
       "Count_ATAC143_CB_FR047.1_ND_sB    ND       \n",
       "Count_ATAC140_CB_FR046.1_ND_sB    ND       \n",
       "Count_ATAC194_CB_FR064.1_ND_sB    ND       \n",
       "Count_ATAC123_CB_FR040.1_ND_sB    ND       \n",
       "Count_ATAC191_CB_FR063.1_ND_sB    ND       \n",
       "Count_ATAC185_CB_FR061.1_ND_sB    ND       \n",
       "Count_ATAC224_CB_FR075.1_ND_sB    ND       \n",
       "Count_ATAC111_CB_FR036.1_NFKB1_sB NFKB1    \n",
       "Count_ATAC295_BG_FR100.1_NFKB1_sB NFKB1    \n",
       "Count_ATAC301_BG_FR102.1_NFKB1_sB NFKB1    \n",
       "Count_ATAC074_CB_FR023.1_NFKB1_sB NFKB1    \n",
       "Count_ATAC304_BG_FR103.1_NFKB1_sB NFKB1    \n",
       "Count_ATAC298_BG_FR101.1_NFKB1_sB NFKB1    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "coldata"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "043d555a-6c47-415d-8257-c31e38c8d378",
   "metadata": {},
   "source": [
    "DESeq2 call significant different peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "6dc06961-209e-4989-a68f-1a1505b3d64c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "class: DESeqDataSet \n",
       "dim: 88870 24 \n",
       "metadata(1): version\n",
       "assays(1): counts\n",
       "rownames(88870): chr1:10043-10267 chr1:180704-180974 ...\n",
       "  chrY:56837247-56837352 chrY:56870861-56871030\n",
       "rowData names(0):\n",
       "colnames(24): Count_ATAC188_CB_FR062.1_ND_sB\n",
       "  Count_ATAC158_CB_FR052.1_ND_sB ... Count_ATAC304_BG_FR103.1_NFKB1_sB\n",
       "  Count_ATAC298_BG_FR101.1_NFKB1_sB\n",
       "colData names(1): condition"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dds <- DESeqDataSetFromMatrix(countData = sample_count_tbl,\n",
    "                              colData = coldata,\n",
    "                              design = ~ condition)\n",
    "dds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "e209594c-6638-4f4d-a4ff-f32d01450522",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in DESeq(dds, contrasts = c(\"NFKB1\", \"ND\")): unused argument (contrasts = c(\"NFKB1\", \"ND\"))\n",
     "output_type": "error",
     "traceback": [
      "Error in DESeq(dds, contrasts = c(\"NFKB1\", \"ND\")): unused argument (contrasts = c(\"NFKB1\", \"ND\"))\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "dds <- DESeq(dds)\n",
    "res <- results(dds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "9e636db1-f515-4112-830c-ba00886eb09f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "out of 88870 with nonzero total read count\n",
      "adjusted p-value < 0.1\n",
      "LFC > 0 (up)       : 5347, 6%\n",
      "LFC < 0 (down)     : 4350, 4.9%\n",
      "outliers [1]       : 18, 0.02%\n",
      "low counts [2]     : 5166, 5.8%\n",
      "(mean count < 3)\n",
      "[1] see 'cooksCutoff' argument of ?results\n",
      "[2] see 'independentFiltering' argument of ?results\n",
      "\n"
     ]
    }
   ],
   "source": [
    "summary(res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "df58ebd1-260a-4bab-8aa8-ce988c95fa58",
   "metadata": {},
   "outputs": [],
   "source": [
    "diff_peaks <- as.data.frame(subset(res, padj < 0.1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "3e02cfa5-7953-4729-94b2-2f492e101aa4",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(diff_peaks, file=\"NFKB1_vs_ND_padj_0.1_Peaks.csv\", sep=\"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "40ab0b4a-4e82-4a2b-aed9-44255ecfa997",
   "metadata": {},
   "source": [
    "Down regulated in NFKB1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "bd252dca-5055-4d64-8e54-2f5d5e4068b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "down_peaks <- diff_peaks[diff_peaks$log2FoldChange < 0, ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "0c26f698-3b23-4df6-9c64-4d9b5a18850b",
   "metadata": {},
   "outputs": [],
   "source": [
    "down_peak_bed <- t(\n",
    "    as.data.frame(\n",
    "        map(rownames(down_peaks), function(x) strsplit(x, \":|-\"))\n",
    "    )\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "32167e86-d9bd-4e06-9ca6-24d7a903865d",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(down_peak_bed, file=\"NFKB1_vs_ND_padj_0.1_NFKB1_Down_Peaks.csv\", row.names = FALSE, col.names = FALSE, sep=\"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8adf61cc-2d55-4f8c-abcc-750f73fd7a6f",
   "metadata": {},
   "source": [
    "Call motif use homer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "98b5537c-61dc-451a-9e65-834f9dfddce5",
   "metadata": {},
   "outputs": [],
   "source": [
    "system2(\n",
    "    \"findMotifsGenome.pl\",\n",
    "    args = c(\"NFKB1_vs_ND_padj_0.1_NFKB1_Down_Peaks.csv\", \"hg38\", \"NFKB1_vs_ND_padj_0.1_NFKB1_Down_Peaks.homer_Motif/\", \n",
    "             \"-size\", \"200\", \"-mask\", \"-preparsedDir\", \"delete\", \"-p\", \"15\"),\n",
    "    stdout=\"call_motif.log\",\n",
    "    stderr=\"call_motif.log\",\n",
    "    wait=TRUE,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "67d25861-de10-4da9-93be-884abd799b64",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "wgcna",
   "language": "R",
   "name": "wgcna"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
